lib/src/components/NxTransferList/NxTransferList.tsx

```diff
@@ -0,0 +1,170 @@
+
+  const availableItemsCountFormatter = availableItemsCountFormatterProp || defaultAvailableItemsCountFormatter,
+      selectedItemsCountFormatter = selectedItemsCountFormatterProp || defaultSelectedItemsCountFormatter;
+
+  const groupedItems = useMemo(() => groupBy(item => selectedItems.has(item.id) ? 'selected' : 'available', allItems),
+          [allItems, selectedItems]),
+
+      // given filter text, returns a function which checks whether a displayName matches the filter.
+      // This returned function will either be the provided `filterFn` (partially applied over the current filter text),
+      // or a case-insensitive substring match
+      mkDisplayNameFilter = (filterText: string) =>
+        filterFn ? partial(filterFn, [filterText]) : pipe(toLower, includes(toLower(filterText))),
+
+      // Given filter text, return a function that filters a list of items
+      mkListFilter = (filterText: string) => filter<DataItem<T>>(
+          pipe(prop('displayName'), mkDisplayNameFilter(filterText))
+      ),
+
+      // The functions to filter each group of inputs
+      availableItemsFilterFn: FilterFn<T> = availableItemsFilter ? mkListFilter(availableItemsFilter) : identity,
+      selectedItemsFilterFn: FilterFn<T> = selectedItemsFilter ? mkListFilter(selectedItemsFilter) : identity,
+
+      // Do the actual filtering, but memoize it since it could be expensive
+      visibleAvailableItems = useMemo(
+          () => availableItemsFilterFn(groupedItems.available || []),
+          [allItems, selectedItems, availableItemsFilter, filterFn]
+      ),
+      visibleSelectedItems = useMemo(
+          () => selectedItemsFilterFn(groupedItems.selected || []),
+          [allItems, selectedItems, selectedItemsFilter, filterFn]
+      );
+
+  const availableCount = allItems.length - selectedItems.size,
+      selectedCount = selectedItems.size;
+
+  function onChange(checked: boolean, id: T) {
+    const newSelectedItems = new Set(selectedItems);
+
+    newSelectedItems[checked ? 'add' : 'delete'](id);
+
+    onChangeProp(newSelectedItems);
+  }
+
+  function onSelectAll() {
+    const newSelectedItems = new Set<T>(selectedItems);
+    for (const item of visibleAvailableItems) {
+      newSelectedItems.add(item.id);
+    }
+
+    onChangeProp(newSelectedItems);
+  }
+
+  function onUnselectAll() {
+    const newSelectedItems = new Set<T>(selectedItems);
+    for (const item of visibleSelectedItems) {
+      newSelectedItems.delete(item.id);
+    }
+
+    onChangeProp(newSelectedItems);
+  }
+
+  return (
+    <div className={classnames('nx-transfer-list', classNameProp)} { ...attrs }>
+      <NxFieldset className="nx-transfer-list__half" label={availableItemsLabel || 'Available Items'}>
+        <div className="nx-transfer-list__control-box">
+          <NxFilterInput className="nx-transfer-list__filter"
+                         value={availableItemsFilter}
+                         onChange={onAvailableItemsFilterChange} />
```

Comment:
Should we add a placeholder? I think the mockups had a "Filter" placeholder.

Reply:
> Should we add a placeholder? I think the mockups had a "Filter" placeholder.

Good question.  Are you suggesting just hardcoding it to "Filter" or having it be configurable?  I think the former might be adequate

Reply:
> I think the former might be adequate

b6bebb7

Reply:
>  Are you suggesting just hardcoding it to "Filter" or having it be configurable? I think the former might be adequate

@alexalyons126 has confirmed that hardcoding "Filter" as the placeholder is adequate